src/pages/profile.tsx

```diff
@@ -0,0 +1,103 @@
+import React, {FormEvent, useState} from 'react';
+import styles from "../components/app/styles.module.css";
+import AppHeader from "../components/app-header/app-header";
+import {EmailInput, Input, PasswordInput} from "@ya.praktikum/react-developer-burger-ui-components";
+import Link from "../components/link/link";
+import {routes} from "./index";
+
+import stylesProfilePage from "./profile.module.css";
+import {useDispatch, useSelector} from "react-redux";
+import {AppDispatch, RootState} from "../services";
+import {logoutUser, updateUser} from "../services/user";
+
+
+const Profile = () => {
+    const dispatch = useDispatch<AppDispatch>()
+    const {user} = useSelector((state: RootState) => state.userStore)
+
```

Comment:
Можно лучше: стоит типизировать хуки `useDispatch` и `useSelector` и использовать уже типизированные версии. Тогда дополнительно прокидывать ничего не нужно будет


```ts
import { useDispatch, useSelector, useStore } from 'react-redux'
import type { AppDispatch, AppStore, RootState } from './store'

// Use throughout your app instead of plain `useDispatch` and `useSelector`
export const useAppDispatch = useDispatch.withTypes<AppDispatch>()
export const useAppSelector = useSelector.withTypes<RootState>()
export const useAppStore = useStore.withTypes<AppStore>()

```

https://react-redux.js.org/using-react-redux/usage-with-typescript#define-typed-hooks